crypto/s2n_rsa.c

```diff
@@ -120,6 +120,7 @@ static int s2n_rsa_encrypt(const struct s2n_pkey *pub, struct s2n_blob *in, stru
     /* Safety: RSA_public_encrypt does not mutate the key */
     int r = RSA_public_encrypt(in->size, (unsigned char *) in->data, (unsigned char *) out->data,
             s2n_unsafe_rsa_get_non_const(pub_key), RSA_PKCS1_PADDING);
+    POSIX_ENSURE(r > 0, S2N_ERR_ENCRYPT);
     POSIX_ENSURE((int64_t) r == (int64_t) out->size, S2N_ERR_SIZE_MISMATCH);
```

Comment:
https://docs.openssl.org/master/man3/RSA_public_encrypt/
> A return value of 0 is not an error and means only that the plaintext was empty.
```suggestion
    POSIX_ENSURE(r >= 0, S2N_ERR_ENCRYPT);
    POSIX_ENSURE((int64_t) r == (int64_t) out->size, S2N_ERR_SIZE_MISMATCH);
```
Check RSA_private_decrypt too, I suspect it's the same.